packages/blue-sdk-viem/src/fetch/v2/VaultV2.ts

```diff
@@ -0,0 +1,162 @@
+  }
+
+  const [
+    token,
+    asset,
+    totalSupply,
+    totalAssets,
+    performanceFee,
+    managementFee,
+    virtualShares,
+    lastUpdate,
+    maxRate,
+    liquidityAdapter,
+    adaptersLength,
+    performanceFeeRecipient,
+    managementFeeRecipient,
+  ] = await Promise.all([
+    fetchToken(address, client, parameters),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "asset",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "totalSupply",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "_totalAssets",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "performanceFee",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "managementFee",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "virtualShares",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "lastUpdate",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "maxRate",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "liquidityAdapter",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "adaptersLength",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "performanceFeeRecipient",
+    }),
+    readContract(client, {
+      ...parameters,
+      address,
+      abi: vaultV2Abi,
+      functionName: "managementFeeRecipient",
+    }),
+  ]);
+
+  const adapters = await Promise.all(
+    Array.from({ length: Number(adaptersLength) }, (_, i) =>
+      readContract(client, {
+        ...parameters,
+        address,
+        abi: vaultV2Abi,
+        functionName: "adapters",
+        args: [BigInt(i)],
+      }),
+    ),
+  );
+
+  return new VaultV2({
+    ...token,
+    totalSupply,
+    totalAssets,
+    asset,
+    performanceFee,
+    managementFee,
+    maxRate,
+    virtualShares,
+    lastUpdate,
+    liquidityAdapter,
+    adapters,
+    performanceFeeRecipient,
+    managementFeeRecipient,
+  });
+}
+
+export async function fetchAccrualVaultV2(
+  address: Address,
+  client: Client,
+  parameters: DeploylessFetchParameters = {},
+) {
+  parameters.chainId ??= await getChainId(client);
+
+  const vaultV2 = await fetchVaultV2(address, client, parameters);
+
+  const [assetBalance, ...adapters] = await Promise.all([
+    readContract(client, {
+      ...parameters,
+      address: vaultV2.asset,
```

Comment:
It would require 1 less RPC call to first fetch the vault's asset and adapters count and then parallelize the rest (2 calls vs 3 currently)

But I guess we don't really care as of now while implementing this is not trivial